### PR TITLE
Training Form 

### PR DESCRIPTION
**Pull request by sylviamclaughlin**
# Summary | Résumé

Initial implementation of training form. Major changes added:

- Upgraded to Bootstrap 5
- Changed the layout of the screens to be more aesthetically pleasant with the extra options. 
- Added the Training form DB models
- Added the Training form Forms
- Added methods for users to be able to submit a training form and view submitted training forms. 
- Still to do: Translation, deletion/changing of training form, unit tests & PDF generation

Screenshot 2023-11-07 at 4 15 17 PM

Screenshot 2023-11-07 at 4 17 10 PM

Screenshot 2023-11-07 at 4 17 26 PM

Screenshot 2023-11-07 at 4 15 29 PM

Screenshot 2023-11-07 at 4 16 04 PM

Screenshot 2023-11-07 at 4 16 38 PM

### CODE REVIEW

File: saas_app/training_request/models.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from django.contrib.auth.models import User
+from user.models import Users
+from internal_ops.models import FundCenter
+from submit_request.models import Currency
+
+
+# DB Model for the course
+class Course(models.Model):
+    title = models.CharField(max_length=100)
+    description = models.CharField(max_length=100)
```
Comment (wanpengyang) [inline saas_app/training_request/models.py:11]: The description could get long for some courses. Can we increase the max length?

**Review by wanpengyang — APPROVED** [review]
👍 awesome work!

File: saas_app/templates/index.html [hunk 2]
```diff
@@ -10,28 +10,64 @@
...
+       <h5>Training Requests</h5> 
+      </button>
+    </h2>
+    <div id="collapseTwo" class="accordion-collapse collapse" aria-labelledby="headingTwo" data-bs-parent="#accordionExample">
+      <div class="accordion-body">
+       <div class="card">
+        <div class="card-body">
+          <h5 class="card-title">{% trans "Submit a Training Request Form" %}</h5>
+          <p class="card-text">{% trans "Please fill out this form if you would like to request training or courses related to your job." %}</p>
+          <a href="training_request" class="btn btn-primary">{% trans "Submit training request" %}</a>
```
[the remark this answers is not in the document]
Comment (gcharest) [reply saas_app/templates/index.html:52]: image

**Comment by gcharest** [comment]
The start date fields requires a date but the user can enter anything, the validation occurs on submission:
image

It also doesn't specify what a valid date would be

**Review by gcharest — COMMENTED** [review]
Not blocking but probably should be looked into

File: saas_app/training_request/models.py (hunk 1 above)
Comment (sylviamclaughlin) [reply saas_app/training_request/models.py]: Thanks for noticing Wanpeng! I have changed now the length to 2000.

File: saas_app/templates/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (sylviamclaughlin) [reply saas_app/templates/index.html:52]: Great suggestion Guillaume! I have added this CSS to the styles.css file.

**Comment by sylviamclaughlin** [comment]
> The start date fields requires a date but the user can enter anything, the validation occurs on submission: image
> 
> It also doesn't specify what a valid date would be

Yes! I will work on getting this sorted and adding a calendar icon in the next PR.

**Pull request merged**